dbms/src/Storages/DeltaMerge/tests/gtest_segment_test_basic.cpp

```diff
@@ -0,0 +1,430 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <Common/CurrentMetrics.h>
+#include <DataStreams/OneBlockInputStream.h>
+#include <Storages/DeltaMerge/DMContext.h>
+#include <Storages/DeltaMerge/DeltaMergeStore.h>
+#include <Storages/DeltaMerge/Segment.h>
+#include <Storages/DeltaMerge/tests/DMTestEnv.h>
+#include <Storages/DeltaMerge/tests/gtest_segment_test_basic.h>
+#include <Storages/Transaction/TMTContext.h>
+#include <Storages/tests/TiFlashStorageTestBasic.h>
+#include <TestUtils/TiFlashTestBasic.h>
+
+namespace DB
+{
+namespace DM
+{
+namespace tests
+{
+void SegmentTestBasic::reloadWithOptions(SegmentTestOptions config)
+{
+    TiFlashStorageTestBasic::SetUp();
+    options = config;
+    table_columns = std::make_shared<ColumnDefines>();
+
+    root_segment = reload(config.is_common_handle);
+    ASSERT_EQ(root_segment->segmentId(), DELTA_MERGE_FIRST_SEGMENT_ID);
+    segments.clear();
+    segments[DELTA_MERGE_FIRST_SEGMENT_ID] = root_segment;
+}
+
+PageId SegmentTestBasic::createNewSegmentWithSomeData()
+{
+    SegmentPtr new_segment;
+    std::tie(root_segment, new_segment) = root_segment->split(dmContext(), tableColumns());
+
+    const size_t num_rows_write_per_batch = 100;
+    {
+        // write to segment and flush
+        Block block = DMTestEnv::prepareSimpleWriteBlock(0, num_rows_write_per_batch, false);
+        new_segment->write(dmContext(), std::move(block), true);
+    }
+    {
+        // write to segment and don't flush
+        Block block = DMTestEnv::prepareSimpleWriteBlock(num_rows_write_per_batch, 2 * num_rows_write_per_batch, false);
+        new_segment->write(dmContext(), std::move(block), false);
+    }
+    return new_segment->segmentId();
+}
+
+size_t SegmentTestBasic::getSegmentRowNumWithoutMVCC(PageId segment_id)
+{
+    auto segment = segments[segment_id];
+    auto in = segment->getInputStreamRaw(dmContext(), *tableColumns());
+
+    size_t num_rows_read = 0;
+    in->readPrefix();
+    while (Block block = in->read())
+    {
+        num_rows_read += block.rows();
+    }
+    in->readSuffix();
+    return num_rows_read;
+}
+
+size_t SegmentTestBasic::getSegmentRowNum(PageId segment_id)
+{
+    auto segment = segments[segment_id];
+    auto in = segment->getInputStream(dmContext(), *tableColumns(), {segment->getRowKeyRange()});
+
+    size_t num_rows_read = 0;
+    in->readPrefix();
+    while (Block block = in->read())
+    {
+        num_rows_read += block.rows();
+    }
+    in->readSuffix();
+    return num_rows_read;
+}
+
+void SegmentTestBasic::checkSegmentRow(PageId segment_id, size_t expected_row_num)
```

Comment:
I can't find the usage place of this function. Do we automatically check the correctness?